packages/flutter/lib/src/widgets/router.dart

```diff
@@ -1146,6 +1159,19 @@ abstract class RouteInformationProvider extends ValueListenable<RouteInformation
   void routerReportsNewRouteInformation(RouteInformation routeInformation) {}
 }
 
+/// This class is used by the [PlatformRouteInformationProvider] to carry
+/// additional restoration information from the platform.
+class _PlatformRouteInformation extends RouteInformation {
```

Comment:
Why do we need this class if `RouteInformation` already has a field for restorationData?